Source/hydro/ppm.cpp

```diff
@@ -0,0 +1,157 @@
+#include "Castro.H"
+#include "Castro_F.H"
+#include "Castro_hydro_F.H"
+
+#ifdef RADIATION
+#include "Radiation.H"
+#endif
+
+#include <cmath>
+#include <ppm.H>
+
+using namespace amrex;
+
+AMREX_GPU_HOST_DEVICE
+void
+Castro::ppm_reconstruct(const Real* s,
+                        const Real flatn, Real& sm, Real& sp) {
+
+  // This routine does the reconstruction of the zone data into a parabola.
+
+  // Compute van Leer slopes
+
+  Real dsl = 2.0_rt * (s[im1] - s[im2]);
+  Real dsr = 2.0_rt * (s[i0] - s[im1]);
+
+  Real dsvl_l = 0.0_rt;
+  if (dsl*dsr > 0.0_rt) {
+    Real dsc = 0.5_rt * (s[i0] - s[im2]);
+    dsvl_l = std::copysign(1.0_rt, dsc) * std::min(std::abs(dsc),std::min(std::abs(dsl),std::abs(dsr)));
+  }
+
+  dsl = 2.0_rt * (s[i0] - s[im1]);
+  dsr = 2.0_rt * (s[ip1] - s[i0]);
+
+  Real dsvl_r = 0.0_rt;
+  if (dsl*dsr > 0.0_rt) {
+    Real dsc = 0.5_rt * (s[ip1] - s[im1]);
+    dsvl_r = std::copysign(1.0_rt, dsc) * std::min(std::abs(dsc),std::min(std::abs(dsl),std::abs(dsr)));
+  }
+
+  // Interpolate s to edges
+
+  sm = 0.5_rt * (s[i0] + s[im1]) - (1.0_rt/6.0_rt) * (dsvl_r - dsvl_l);
+
+  // Make sure sedge lies in between adjacent cell-centered values
+
+  sm = std::max(sm, std::min(s[i0], s[im1]));
```

Comment:
On the GPU we need to use amrex::min and amrex::max instead of std::min and std::max.

Reply:
fixed